src/Workspaces/Core/Portable/Workspace/Solution/RecoverableTextAndVersion.cs

```diff
@@ -89,14 +88,19 @@ public bool TryGetTextVersion(LoadTextOptions options, out VersionStamp version)
             return false;
         }
 
-        public TextAndVersion GetValue(LoadTextOptions options, CancellationToken cancellationToken)
+        private async ValueTask<RecoverableText> GetRecoverableTextAsync(
+            bool useAsync, LoadTextOptions options, CancellationToken cancellationToken)
```

Comment:
extracted common helper as we already had the code duplicated in two methods and i didn't want it in a third.  Has to have 'ValueTask + useAsync' pattern as the two existing codepaths are the same concept, except one is the sync path and the other async path.

Reply:
This has come up a few times. It would be nice to have a clean standard practice for the approach. My main concern here is the need for a CA2012 suppression. I'm curious if we can come up with another approach that doesn't need that. Perhaps we have a type that is effectively `(T SynchronousResult, ValueTask<T> AsynchronousResult)`, but with less difficulty in using it (i.e. shorter name, one generic argument).

Reply:
It could have a static `FromResult` method like this:

```csharp
static WhateverType<T> FromResult(bool useAsync, T result)
  => useAsync ? new WhateverType<T>(new ValueTask<T>(result)) : new WhateverType<T>(result);
```

Reply:
Happy to adopt a pattern of you want to make it :-)